### ISSUE TITLE
[Rule Help] Create Documentation for style_hover_persistent

### ISSUE THREAD

**Issue by mbgower** [IBMa/equal-access#625]
Generate help content and other documentation when Implementation of #623 matures enough
- [x] confirm messages
- [x] create rule help
Documentation at https://ibm.ent.box.com/file/521562780202

**Comment by drjoho** [comment]
// Trigger logic:
        //
        // First, logic for the sibling failures
        // 
        // 1. Do we have an element with a hover selector,
        //    e.g., a:hover, span:hover, etc...?
        //
        // 2. is the hover followed by either the + or ~ css combinators?
        //
        //    e.g., :hover+ or :hover~ (after triming spaces)
        //
        // 3. is the element after the css combinator + or ~ followed by an element 
        //    with a display attribute with any value but none? e.g, 
        //
        //    [element1]:hover + [element2] {
        //       display: block;
        //
        //    [element1]:hover ~ [element2] {
        //       display: block;
        //
        // 4. if we found + and there is one sibling directly after [element1] in the body
        //    of type [element2]. 
        //    if 1-4 are true Trigger Failure 1
        //
        // 5. if we found ~ and there are two or more siblings directly after [element1] of type
        //    [element2] - note there cannot be other elements inbetween the siblings. 
        //    if 1-3 and 5 are true Trigger Failure 2

**Comment by mbgower** [comment]
rpt-style-rules.ts

**Comment by mbgower** [comment]
Suggested passive0 messages options:
- brief
  > The pointer should be able to move over content displayed on hover

- more context
  > When content displays on hover, the pointer should be able to move from the trigger to the new content

**Comment by mbgower** [comment]
Suggested active messages:
1. The pointer can hover on the displayed element, not just the trigger
1. Only the element adjacent to the trigger will remain visible when multiple sibling elements are displayed on hover
1. The `margin` style attribute on the displayed element causes the content to disappear when the pointer is moved from the trigger to the new content

**Comment by mbgower** [comment]
Suggested text to include in the help for each scenario
1. The pointer can hover on the displayed element, not just the trigger
1. Only the element adjacent to the trigger will remain visible when multiple sibling elements are displayed on hover
1. The `margin` style attribute on the displayed element causes the content to disappear when the pointer is moved from the trigger to the new content

**Comment by mbgower** [comment]
Committed messages and help file as commits in the existing https://github.com/IBMa/equal-access/tree/joho-hoverRule-623 branch

**Comment by mbgower** [comment]
Updated help file, including adding what is likely an error-prone attempt at a first code sample, and committed to the branch. 
@philljenkins I'd like to run over this draft with you.

@josiah, note that my

**Comment by mbgower** [comment]
Added a commit for the second and third potential messages to ...-engine/src/v2/checker/accessibility/help/index.ts

Confirmed that rulesets/index.ts already looks properly built

**Comment by mbgower** [comment]
Created word doc of text changes and passed on to phill for review.

**Comment by philljenkins** [comment]
Sent review comments on messages and help content to @mbgower

**Comment by philljenkins** [comment]
@mbgower @drjoho can a \"{0}\" token that contains the element type, label, or ID of the element which is triggering the hover be added to the potential failure messages? That would aid the user in identifying the specific occurrence?  
>"Potential_1": "Confirm the pointer can be positioned over the displayed **_\"{0}\"_** element, not just the trigger",
>"Potential_2": "Confirm the pointer can be positioned over _all_ the information displayed **_in \"{0}\"_** on hover",
>"Potential_3": "Confirm the **'**margin**'** style attribute on the displayed **_\"{0}\"_** element has not prevented the pointer from hovering over the displayed content, not just the trigger"

**Comment by mbgower** [comment]
@philljenkins 
That's certainly worth considering, and we can add it as an iteration, but there's no reason we can't implement this rule without it in the first version

**Comment by mbgower** [comment]
Committed some changes based on  feedback from Phill.
Any updates due to formatting can be dealt with as part of the rule implementation issue.
